### PR TITLE
Bump to Chisel 3.5

### PR DESCRIPTION
**Pull request by davidbiancolin**
This pulls in all the necessary changes from currently open PRs to complete a bump to Chisel 3.5. For various reasons, some of those earlier PRs cannot pass CI standalone, so it was easiest (for me) to bundle them together. 
These PRs include:
Bumping hardfloat (closes #2930)
Removing clonetype (closes #2889)
Removing the old property and updating cover (closes #2890)

Other notes: 
- I bumped macro paradise to 2.1.1 so that a published version could be found for scala 2.12.15. We already do this in the mill build, it appears. If that is undesired, can someone point me at a place where it is published and we can update the resolvers?
- I tried to update the mill files. @sequencer how should i test them?

**Related issue**: 

**Type of change**: other enhancement

**Impact**: API addition (no impact on existing code)

**Development Phase**:  implementation

**Release Notes**

Bump to Chisel 3.5.

### CODE REVIEW

**Comment by sequencer** [comment]
Thanks @davidbiancolin! I'll work check this today(maybe 8~10 hours later)

**Comment by davidbiancolin** [comment]
> Thanks @davidbiancolin! I'll work check this today(maybe 8~10 hours later)

Feel free to just push to this branch.

**Review by jackkoenig — APPROVED** [review]

File: common.sc [hunk 1]
```diff
@@ -4,8 +4,8 @@ import mill.scalalib.publish._
 import coursier.maven.MavenRepository
 
 val defaultVersions = Map(
-  "chisel3" -> "3.4.3",
-  "chisel3-plugin" -> "3.4.3"
+  "chisel3" -> "3.5.0",
+  "chisel3-plugin" -> "3.5.0"
```
Comment (sequencer) [inline common.sc:8]: Approve after 
https://github.com/sequencer/playground/actions/runs/1710452837
passes

**Review by sequencer — APPROVED** [review]

**Comment by sequencer** [comment]
Finally!

**Comment by jackkoenig** [comment]
@michael-etzkorn you definitely need the plugin to use the same (at least major) version of Chisel. All Chisel related projects must be in sync on major version, see https://www.chisel-lang.org/chisel3/docs/appendix/versioning.html

**Pull request merged**